master/docs/manual/configuration/reporters.rst

```diff
@@ -785,6 +787,236 @@ Two additional arguments can be set to control how fast the IRC bot tries to rec
+
+    A sample ``authz`` parameter may look as follows:
+
+    .. code-block:: python
+
+        authz=(
+          'getid': True,
+          '': [123456, 789012],
+          ('force', 'stop'): [123456],
+        )
+
+    Anybody will be able to run the ``getid`` command, users with IDs 123456 and 789012 will be allowed to run any safe command and the user with ID 123456 will also have the right to force and stop builds.
+
+``tags``
+    (optional)
+    When set, this bot will only communicate about builders containing those tags.
+    (tags functionality is not yet implemented)
+
+``notify_events``
+    (optional)
+    A list or set of events to be notified on the Telegram chats.
+    Telegram bot can listen to build 'start' and 'finish' events. It can also notify about missing workers and their return.
+    This parameter can be changed during run-time by sending the ``/notify`` command to the bot.  Note however, that at the buildbot restart or reconfig the notifications listed here will be turned on for the specified chats. On the other hand, removing events from this parameters will not automatically stop notifications for them (you need to turn them off for every channel with the ``/notify`` command).
+
+``showBlameList``
+    (optional, disabled by default)
+    Whether or not to display the blame list for failed builds.
+    (blame list functionality is not yet implemented)
+
+``useRevisions``
+    (optional, disabled by default)
+    Whether or not to display the revision leading to the build the messages are about.
+    (useRevisions functionality is not yet implemented)
+
+``certificate``
+    (optional)
+    A content of the SSL certificate of a certificate file open in binary mode. This is necessary if the access to the Buildbot web interface is through HTTPS protocol with self-signed certificate and ``userPolling`` is not set to ``True``.
+
+
+``usePolling``
+    (optional, disabled by default)
+    By default this bot receives messages from Telegram through web-hooks. However, this requires the web frontend of the Buildbot to be configured and accessible through HTTPS (not HTTP) on a public IP and port number 443, 80, 88, or 8443. Furthermore, the Buildbot configuration option :bb:cfg:`buildbotURL` must be correctly set. If this is not possible, :bb:reporter:`TelegramBot` can use polling to receive messages from Telegram. Set ``usePolling`` to ``True`` to do so.
+
+``pollTimeout``
+    (optional)
+    The default time the bot should wait for Telegram to respond to polling using `long polling <https://en.wikipedia.org/wiki/Push_technology#Long_polling>`.
```

Comment:
Franckly, long polling is good tech and works well for this kind of stuff. 
For me It is better than webhook, as simpler to setup.
I have lots of experience with Amazon SQS, which works that way, and never had issue with msg lost despite actively looking for them.
Polling tends to have bad reputation, but in this case, I think it is better.

Having two way of connecting sounds like overkill for maintainance, we should choose one.
I would choose long polling for simplicity and efficiency.

I do realize there is quite amount of quality code for web hook mode that I ask you to throw away :-(

Reply:
It does not seem that there's that much additional code to maintain. I think that because the code is already written, has few chance to impact the codebase negatively and has tests, we should just accept it.

Reply:
Well this also adds a precedent, people could think it is a good thing to implement both.
 
We have other support which have both hook or poller. but in all those case, the poller is very inefficient. Here, this is not the case

Reply:
Frankly, I would rather keep the code for web-hook along long polling. There is a number of reasons for this:

1. If you take a look at telegram bot frameworks in the web, there is a tendency to use web-hooks in productions servers: more and more bots are using this approach and it is always better go go with the stream. Also official telegram docs specify that there are advantages of using web-hooks instead of polling.

2. The interface code for both polling and web-hook is quite thin, so it should not be too hard to maintain.

3. It is good to keep the web-hook code (maybe encapsulated and better separated from the telegram-specific stuff) as it will make straightforward to add more bots. The two most obvious that come to my mind are Messenger and Slack.

Reply:
ok

Reply:
I have refactored web-hook code a bit and moved it to words. It should be easier to re-use it in the future...